pkg/machines/libvirt-dbus.es6

```diff
@@ -0,0 +1,861 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Provider for Libvirt using libvirt-dbus API.
+ * See https://github.com/libvirt/libvirt-dbus
+ */
+import cockpit from 'cockpit';
+import $ from 'jquery';
```

Comment:
Please avoid introducing jQuery into new code. It's only being used once for the `isEmptyObject()` function. I'll write there how to replace it.

Reply:
The thing to avoid is to mix React and jQuery UI building (e. g. show/hide elements), as they are both colliding technically and also makes it hard to understand as a developer what a page is doing.

For XML parsing, doesn't [DOMParser](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser) work? It would avoid pulling the sizable jQuery into the page just for that one function. Otherwise, if DOMParser does not work, using jQuery for XML parsing would be acceptable.

Reply:
DOMParser works, I replaced all occurrences in machines code.